classes/task/report_async.php

```diff
@@ -0,0 +1,94 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * report_coursesize tasks
+ *
+ * @package   report_async
+ * @copyright Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+namespace report_coursesize\task;
+
+/**
+ * report_async tasks
+ *
+ * @package   report_async
+ * @copyright Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class report_async extends \core\task\scheduled_task {
+
+    /**
+     * Get task name
+     */
+    public function get_name() {
+        return get_string('pluginname', 'report_coursesize');
+    }
+
+    /**
+     * Execute task
+     */
+    public function execute() {
+        global $DB;
+
+        mtrace("Generating report_coursesize cache...");
+
+        set_time_limit(0);
+
+        // Clean up cache table.
+        $DB->delete_records('report_coursesize');
```

Comment:
worth a transaction?

Reply:
yeah - I was thinking about that at the time I was writing this but didn't come back to it - I'll add one in.